src/modules/routing/utils/getPath.js

```diff
@@ -37,7 +37,9 @@ const getPath = (to, routes) => {
 	const { params = {}, query = {} } = to;
 	let queryStringParts = [];
 	each(params, (paramValue, paramKey) => {
-		const replaced = result.replace(new RegExp(`:${paramKey}(?=\/|$)`), paramValue);
+		const replaced = !paramValue ?
+			result.replace(new RegExp(`/:${paramKey}\\?(?=\/|$)`), '') :
```

Comment:
pokud neni nastavena hodnota `paramValue`, nahradit pouze nepovinny parametr (ma na konci `?`) za nic (`''`). Pokud je parametr povinny, chova se to stejne jako pred tim: `result = replaced`.